### PR TITLE
Replacing fatal with error for better logging error.

### PR DESCRIPTION
**Pull request by mourya007**
ref: https://github.com/kubernetes/kubernetes/pull/72350/files#r244227792

### CODE REVIEW

**Comment by mourya007** [comment]
cc @logicalhan

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic_test.go [hunk 1]
```diff
@@ -279,12 +279,12 @@ func buildTestHandler(t *testing.T, a *atomic.Value) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		body, err := ioutil.ReadAll(r.Body)
 		if err != nil {
-			t.Fatalf("could not read request body: %v", err)
+			t.Errorf("could not read request body: %v", err)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic_test.go]: what's the value of `body` then?

File: staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/webhook/webhook_test.go [hunk 2]
```diff
@@ -486,7 +486,7 @@ func TestWebhookTokenAuthenticator(t *testing.T) {
 		t.Run(tt.description, func(t *testing.T) {
 			wh, err := newTokenAuthenticator(s.URL, clientCert, clientKey, caCert, 0, tt.implicitAuds)
 			if err != nil {
-				t.Fatal(err)
+				t.Error(err)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/webhook/webhook_test.go]: what's the value of `wh`?

**Comment by mourya007** [comment]
@sttts  Thanks for the review, Motive of this change is that when we use fatal it will gives os.exit(1) at the end. So I think error can be use here instead.

**Comment by sttts** [comment]
> Motive of this change is that when we use fatal it will gives os.exit(1) at the end. So I think error can be use here instead.

`t.Fatal` stops the unit test, not the binary.

You cannot just switch Fatals to Errors without making sure the test does not panic afterwards.

**Pull request closed without merging**